### PR TITLE
LPD-15357 Create test coverage reindex own instance

### PR DESCRIPTION
**Pull request by gustavolimav**
https://liferay.atlassian.net/browse/LPD-15357

I had convert the unit tests to integration tests following this request from Bchan https://github.com/brianchandotcom/liferay-portal/pull/146849

### CODE REVIEW

**Review by joshuacords — CHANGES_REQUESTED** [review]
Our portal-search-admin-web modules already has tests within it, so I don't see a need to create a new portal-search-admin-web-test. Could you move these test into the existing module under a testIntegration folder? (See journal-test which has both unit and integration tests)

**Comment by gustavolimav** [comment]
Hi @joshuacords, I haven't found any example of a web module that has integration tests inside it. I'm following the pattern I found in the Liferay code, I haven't found a reason why it's like that, but I figured it had a reason. Do you have any example of that?

As I explained in the PR that wasn't related to this story, the user can't have access to this class if they can't have access to the search portlet in the control panel, so the Feature flag here is unnecessary, without the feature flag only omniadmins will have access to the search portlet in contral panel, with the feature flag maybe some company admins will have access. In general, I don't think adding the resource flag to this class makes sense. The only reason I see is to test the actual behavior and the behavior we're adding in this story. I can see benefits in having it explicit in the code and in the test. I can add it, it's not a big deal, just to make it clear that the absence of the feature flag is not a problem in terms of functionality.

And @lipusz, could you please don't close the PR before we conclude the conversation? I could still continue pushing code here instead of open a new one, I can't see benefit in close it.

**Comment by joshuacords** [comment]
Hi @xbrianlee,
Could you look at the last 2 comments of this PR where we discuss the location of tests in modules. Would you happen to know if there's a liferay standard to follow?

**Comment by xbrianlee** [comment]
@joshuacords Unfortunately I'm not familiar with any criteria for the location of tests in modules. If I can use functional test organization as a proxy, I'd say there are likely multiple standards that teams follow. Every other year or so tests get reorganized as new features and modules get added.

**Comment by joshuacords** [comment]
> As I explained in the PR that wasn't related to this story, the user can't have access to this class if they can't have access to the search portlet in the control panel, so the Feature flag here is unnecessary, without the feature flag disabled only omniadmins will have access to the search portlet in contral panel, with the feature flag enabled maybe some company admins will have access depending on the configuration. In general, I don't think adding the feature flag to this class makes sense. The only reason I see is to test the actual behavior and the behavior we're adding in this story. I can see benefits in having it explicit in the code and in the test. I can add it, it's not a big deal, just to make it clear that the absence of the feature flag is not a problem in terms of functionality.

I manually tested this to confirm - a feature flag check isn't necessary in the class or in the test because access is already blocked by the portal.

**Comment by joshuacords** [comment]
I tried running the integration tests, both with and without the feature flag enabled, and it's failing:

> com.liferay.portal.search.admin.web.internal.portlet.action.test.EditMVCActionCommandTest STANDARD_OUT
>     Loading jar:file:/home/me/dev/bundles/LPD-6591-2/tomcat-9.0.83/webapps/ROOT/WEB-INF/shielded-container-lib/portal-impl.jar!/system.properties
>     Loading jar:file:/home/me/dev/projects/liferay-portal/.m2/com/liferay/portal/com.liferay.portal.impl/98.2.0-SNAPSHOT/com.liferay.portal.impl-98.2.0-SNAPSHOT.jar!/system.properties
> 
> com.liferay.portal.search.admin.web.internal.portlet.action.test.EditMVCActionCommandTest > classMethod STARTED
> 
> com.liferay.portal.search.admin.web.internal.portlet.action.test.EditMVCActionCommandTest > classMethod FAILED
>     java.net.ConnectException: Connection refused (Connection refused)
>         at java.net.PlainSocketImpl.socketConnect(Native Method)
>         at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
>         at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
>         at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
>         at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
>         at java.net.Socket.connect(Socket.java:589)
>         at java.net.Socket.connect(Socket.java:538)
>         at java.net.Socket.<init>(Socket.java:434)
>         at java.net.Socket.<init>(Socket.java:244)
>         at com.liferay.arquillian.extension.junit.bridge.client.FrameworkState.connect(FrameworkState.java:53)
>         at com.liferay.arquillian.extension.junit.bridge.client.ClientState.open(ClientState.java:67)
>         at com.liferay.arquillian.extension.junit.bridge.junit.Arquillian.run(Arquillian.java:111)
>         at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
>         at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
>         at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
>         at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
>         at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
>         at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>         at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
>         at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
>         at java.lang.reflect.Method.invoke(Method.java:498)
>         at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
>         at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
>         at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
>         at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
>         at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
>         at org.gradle.api.internal.tasks.testing.worker.TestWorker$2.run(TestWorker.java:176)
>         at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
>         at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
>         at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
>         at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
>         at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:133)
>         at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:71)
>         at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
>         at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
> 
> 1 test completed, 1 failed
> 

@gustavolimav - do you know what the issue may be?

**Comment by joshuacords** [comment]
It appears that the test runs fine when a portal is not running, but I can't get it to run successfully on an already running portal.

**Comment by joshuacords** [comment]
Hi @gustavolimav, sorry for the confusion, besides removing the lines from the build.gradle, this looks good to forward. 

We'll see if B.Chan likes the new module. In the abstract it does seem better to me to keep testing in separate modules where possible to reduce the size of what we're delivering so we'll go with your approach.

**Comment by gustavolimav** [comment]
@joshuacords, what you mean with build.gradle lines? did you submit any review?

**Review by joshuacords — CHANGES_REQUESTED** [review]

File: modules/apps/portal-search/portal-search-admin-web-test/bnd.bnd [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Bundle-Name: Liferay Portal Search Admin Web Test
+Bundle-SymbolicName: com.liferay.portal.search.admin.web.test
```
Comment (joshuacords) [inline modules/apps/portal-search/portal-search-admin-web-test/bnd.bnd:2]: Test Comment

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:search

**Comment by gustavolimav** [comment]
ci:test:relevant

**Comment by gustavolimav** [comment]
ci:report:2536685543

**Comment by gustavolimav** [comment]
ci:report:2536680054

**Comment by gustavolimav** [comment]
Tests looks good, I will forward manually

**Comment by gustavolimav** [comment]
https://github.com/brianchandotcom/liferay-portal/pull/147115

**Pull request closed without merging**